### ISSUE TITLE
[Suggestion] Module Language?

### ISSUE THREAD

**Issue by Fl1pNatic** [absoIute/Mega-Hack-Pro-Future#602]
Would it be cool if you could just, activate a module after a certain time, or at a certain percentage, or when you are in 2 player mode, or when you are using a certain gamemode, stuff like that.

While it might all be really hard to fit all of these in GUI, why don't we make script files, that we can then easily toggle on and off, that we can just pre-make, and use them for as much as we want, without going through troubles of settings it all up in GUI(If it will ever be added).

For example:
```python
if(game.percentage == 50){
    modules.pro.icon_effects.enabled = true;
}
else if(game.percentage < 50){
    modules.pro.icon_effects.enabled = true;
}
```
or
```python
if(game.gamemode == "wave"){
    modules.pro.hitboxes.enabled = true;
}
else{
    modules.pro.hitboxes.enabled = false;
}
```
Something like that

While this might be really hard to do, as it will require making an entire language, or at least a library, it will open up a lot of possibilities. Maybe you could also add extensions support.

This might not get into V7, but if you will start making V8, please consider.

**Comment by MainConMan** [comment]
At a certain percentage would be possible when/if #549 gets implemented. But good ideas! I like them! :)

**Comment by absoIute** [comment]
While I look into implementing this for v7.2 this might be the way I go but not sure yet, it may just have only a visual interface which is a little more restricted but a lot more accessible

**Comment by absoIute** [comment]
True, definitely something I'll look into, along with addressing UI bloat in a future update now it's starting to not look great on 1080p

**Comment by defaultacc30** [comment]
> True, definitely something I'll look into, along with addressing UI bloat in a future update now it's starting to not look great on 1080p

what about module language? like having mhv7 display all mods in Chinese or English UK.

**Comment by absoIute** [comment]
Translation is being considered but would be a lot of work, though for English I will be sticking to UK English, not US English

**Comment by Nicko2102** [comment]
> what about module language? like having mhv7 display all mods in English UK.

different kind of language, but i like the idea

**Comment by defaultacc30** [comment]
> Translation is being considered but would be a lot of work, though for English I will be sticking to UK English, not US English

What about translating for the most used languages? Like russian or spanish

**Comment by Nicko2102** [comment]
he literally answered the question in the comment you replied to.

**Comment by defaultacc30** [comment]
> he literally answered the question in the comment you replied to.

No like, he cant translate into russian or spanish. but he can get his community to do so.

**Comment by Nicko2102** [comment]
I dont want to speak for him but i dont think thats the hard part of translation

**Comment by Underkirby** [comment]
If this was added you could make it so that you have to toggle a hack to enable the ability to toggle hacks (maybe "Allow Extensions to toggle hacks")

**Comment by Nicko2102** [comment]
> If this was added you could make it so that you have to toggle a hack to enable the ability to toggle hacks (maybe "Allow Extensions to toggle modules")

Or have that option for each extension so you can allow only a few if you want

**Comment by Brittank88** [comment]
> Translation is being considered but would be a lot of work

Are there no websites to facilitate crowdsourced translations? I remember seeing that at one point.

**Comment by Brittank88** [comment]
> but he can get his community to do so.

Yeah we'll just forget all about quality assurance
